processing/src/main/java/io/druid/query/aggregation/FilteredAggregatorFactory.java

```diff
@@ -256,8 +269,20 @@ public boolean matches()
       };
     }
 
-    @Override
-    public ValueMatcher makeValueMatcher(final String dimension, final Predicate predicate)
+    public ValueMatcher makeValueMatcher(final String dimension, final DruidPredicateFactory predicateFactory)
+    {
+      ValueType type = getTypeForDimension(dimension);
+      switch (type) {
+        case LONG:
+          return makeLongValueMatcher(dimension, predicateFactory.makeLongPredicate());
+        case STRING:
+          return makeStringValueMatcher(dimension, predicateFactory.makeStringPredicate());
+        default:
+          throw new UOE("Cannot make ValueMatcher for type[%s]", type);
```

Comment:
Similar comment to QueryableIndexStorageAdapter, this should probably act like the column is a string column full of nulls when we cannot filter on that particular column type.